src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -542,12 +642,27 @@ def parse_chat(chat_str, images: List = None, valid_roles: List[str] = None):
 
     for chunk in chunks:
         last_message = chat_list[-1] if len(chat_list) > 0 else None
-        if last_message and "role" in last_message and "content" not in last_message:
+        if is_tool_chunk(last_message):
+            parse_tools(last_message, chunk, hash2images, image_detail)
```

Comment:
Please run an e2e test and add a screenshot in PR description also to show the success run of tools&tool choice.

Reply:
add e2e test and screenshot